refactor(shared-data, api): fold v2 tiprack changes into v1 and remove v2 definitions

jbleon95 opened the pull request:
# Overview

This PR addresses RQA-4403

In #18621 new versions of the flex tiprack definitions were added, in which the only change made was adding a stacking offset for `opentrons_flex_tiprack_lid`. The decision was made to make these a new labware version to enforce a strictness with PAPI-affecting changes, even if they were purely additive and would not change previous behavior.

This change unfortunately is causing issues with liquid classes, since the liquid class definitions are keyed on both pipette model and a single tiprack URI, which includes the version and is currently hardcoded to `1` for version. A few different solutions on the liquid class side were explored, but all of them faced issues in both implementation and timing:
- The naive change of adding in entries for v2 of the tiprack definitions for all liquid classes would both introduce needless redundancy and double the size of the liquid classes from **15,000** to **30,000** lines.
- Various hacking and hardcoding in the PAPI, PD and ODD quicktransfer code which would be bandaids and hacks on this issue, and introduce more code complexity
- Updating the schema to go from a single URI to a list of URIs, which is ultimately the correct solution for this but given the timing of the upcoming release, would not be able to be implemented and tested to our standards, in addition to ensuring backwards compatibility and any update issues

Because none of these solutions were ideal, we looked back into removing the v2 definitions and adding the changes to the existing v1 definitions, and came to the conclusion that this was the least evil of all the options:
- It fixes the liquid class issue in all domains (PAPI, PD, quick transfer) as well as any other place that expects v1 of the tiprack as the only version
- The change made is purely additive and we have precedent of making in-place non-breaking changes to our labware definitions in the past, even if this is something we want to change moving forward
- The changes made are for the stacker, which is new and will only be usable on the latest API version and therefore there would be nor reason to use the new stacking offsets for older protocols, so this further reduces the risk of making this change in place

Therefore we have chosen to go about removing the v2 tiprack definitions and add the stacking offset to v1. This same change was made for the `opentrons_flex_tiprack_lid`, which is also a safe change to made since stacker has yet to be released, and keeps the tipracks and the tiprack lid on similar versions, which are currently tightly coupled for oneline lid loads (this is a separate undesired behavior that should be addressed at some point).

## Changelog

- Remove `2.json` versions of opentrons flex tipracks and tiprack lid
- Add tiprack lid stacking offset to `1.json` for flex tipracks


## Risk assessment

Low.

jbleon95 committed:
fold v2 changes of opentron tiprack defs into v1

jbleon95 committed:
test fix

SyntaxColoring reviewed (approved):
Thanks! Agreed with the rationale, and the implementation looks right.

LGTM once CI passes. I think `ControlledLabwareListItems.test.tsx` has some hard-coded URIs that need to be updated. It looks like you can just downgrade them to v1.

ddcc4 reviewed (commented):
Hm, do you need to completely undo @vegano1's PR #18945 here?

Do you need to roll back the changes to `test_analysis_snapshot*`?

SyntaxColoring committed:
Try to fix opentrons-ai tests.

SyntaxColoring committed:
Update compatible-labware constants.

vegano1 reviewed (approved):
thank you

jbleon95 committed:
Revert "feat(shared-data): update pipette def tiprackURIs to v2 (#18722)"

This reverts commit 5360a4057ad18c74940ca4128fb9bf5508812f92.

SyntaxColoring committed:
Merge branch 'edge' into fold_v2_tiprack_def_changes_into_v1

Merging for an updated analysis snapshot test workflow.

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal fold_v2_tiprack_def_changes_into_v1 snapshots (#18974)

This PR was requested on the PR
https://github.com/Opentrons/opentrons/pull/18970

Co-authored-by: SyntaxColoring <3236864+SyntaxColoring@users.noreply.github.com>
